styleguide.md

```diff
@@ -0,0 +1,180 @@
+
+### Expressions
+
+Different constructs with _expression_ and _call_:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+`if`-expression | `\lstinline!if!-expression` | Generic language concept
+`DynamicSelect`-expression | `\lstinline!DynamicSelect!-expression` | Currently with space in the document!
+`Curve`-expression | `\lstinline!Curve!-expression` | Currently with space in the document!
+`Real` expression | `\lstinline!Real! expression` | An expression of type `Real`
+`y` expression | `\lstinline!y! expression` | Expression for something named `y`
+`convertElement` call | `\lstinline!convertElement! call` | A call expression with callee `convertElement`
+
+Note: There is no need for hyphenation of "`convertElement` call" since we don't say "`Real` call" for a call expression of type `Real` (we have "`Real` expression" for this purpose).
+
+### The keywords themselves
+
+When referencing a keyword itself, hyphenation is not used, and when possible, a better describing word than _keyword_ is used:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+the `input` prefix | `\lstinline!prefix! prefix` | For more prefixes, see `class-prefixes`, `base-prefix`, and `type-prefix` in the grammar.
+the `final` modifier | `the \lstinline!final! modifier` |
+the `each` keyword | `the \lstinline!each! keyword` |
+
+Depending on context, one can also swap the order, or drop the describing word completely:
+
+Appearance | LaTeX source
+--- | ---
+the keyword `each` | `the keyword \lstinline!each!`
+declared as `final` | `declared as \lstinline!final!`
+
+### Named functions and operators
+
+When referencing a named function of operator, hyphenation is not used, and it is common to not combine with any describing word:
+
+Appearance | LaTeX source
+--- | ---
+the `smooth` operator | `the \lstinline!smooth! operator`
+where `pre` is not explicitly used | `where \lstinline!pre! is not explicitly used`
+
+### Other things, special cases
+
+Incomplete list of various terminology with special formatting rules:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+start value | `start value` | Value of the `start`-attribute (there could be exceptions!)
+reduction expression | `reduction expression` |
+base class | `base class` | Similarly: derived class
+base-clock | `base-clock` | Similarly: sub-cock
+
+## Definitions
+
+New terminology is either introduced with a `definition` environment, or as part of the running text.
+When part of the running text, the introduced terminology should be marked with `\firstuse` at the point of the definition.
+(The use of `\firstuse` helps us both produce consistent formatting and keep track of things that should appear in the document index.)
+
+If the new terminology is used before being introduced, the first use should be marked with `\willintroduce` to alert the reader that this is not a term that is expected to be known yet by a first-time reader.
```

Comment:
When I first read that I thought it meant: `\firstuse(some term)\willintroduce`. Maybe making it a bit more explicit?
```suggestion
If the new terminology is used before being introduced, it should be marked with `\willintroduce` (instead of `\firstuse`) to alert the reader that this is not a term that is expected to be known yet by a first-time reader.
```
Or giving an example?

Reply:
Seems like a good suggestion. I don't think an example is needed.